src/api-extension/api-extension.service.ts

```diff
@@ -0,0 +1,114 @@
+import { ConfigService } from '@nestjs/config';
+
+@Injectable()
+export class ApiExtensionService {
+  private client: ByProjectKeyRequestBuilder;
+  constructor(
+    private readonly ctConnector: CommerceToolsConnectorService,
+    private readonly configService: ConfigService,
+  ) {
+    this.client = this.ctConnector.getClient();
+  }
+
+  async list() {
+    const {
+      body: { results: extensions },
+    } = await this.client.extensions().get().execute();
+
+    return extensions;
+  }
+
+  async removeById(id: string) {
+    const extensions = await this.list();
+    const extensionToRemove = extensions.find(
+      (extension) => extension.id === id,
+    );
+
+    if (!extensionToRemove) {
+      return false;
+    }
+    const removedExtenison = (
+      await this.client
+        .extensions()
+        .withId({ ID: extensionToRemove.id })
+        .delete({ queryArgs: { version: extensionToRemove.version } })
+        .execute()
+    ).body;
+
+    return removedExtenison;
+  }
+
+  async removeByAttr(attr: 'key' | 'destination', value: string) {
+    const extensions = await this.list();
+    const extensionsToRemove = extensions.filter(
+      (extension) => extension[attr] === value,
+    );
+    if (!extensionsToRemove?.length) {
+      return false;
+    }
+
+    const removedExtensions: Extension[] = [];
+
+    for (const extension of extensionsToRemove) {
+      removedExtensions.push(
+        (
+          await this.client
+            .extensions()
+            .withId({ ID: extension.id })
+            .delete({ queryArgs: { version: extension.version } })
+            .execute()
+        ).body,
+      );
+    }
+
+    return removedExtensions;
+  }
+
+  async add(url: string, key: string): Promise<string | false> {
+    const authPasswd = this.configService.get<string>(
+      'API_EXTENSION_BASIC_AUTH_PASSWORD',
+    );
+    const body = {
+      destination: {
+        type: 'HTTP' as const,
+        url: `${url}/api-extension`,
+        authentication: {
+          type: 'AuthorizationHeader' as const,
+          headerValue: `Basic ${authPasswd}`,
+        },
+      },
+      triggers: [
+        {
+          resourceTypeId: 'cart' as const,
+          actions: ['Create' as const, 'Update' as const],
+        },
+        {
+          resourceTypeId: 'order' as const,
+          actions: ['Create' as const, 'Update' as const],
+        },
+      ],
+      timeoutInMs: 2000,
+      key,
+    };
+    const addResponse = await this.client.extensions().post({ body }).execute();
+    return addResponse.body?.id || false;
+  }
+
+  async update(url: string): Promise<string | false> {
+    const apiExtensionKey = this.configService.get<string>(
+      'COMMERCE_TOOLS_API_EXTENSION_KEY',
+    );
+
+    if (!apiExtensionKey) {
+      throw new Error(`Missing COMMERCE_TOOLS_API_EXTENSION_KEY configuration`);
+    }
+
+    await this.removeByAttr('key', apiExtensionKey);
```

Comment:
Maybe check if remove was successful and then register new application with given key

Reply:
it's intended to work that way